api/services/ZoomAccount/index.js

```diff
@@ -10,22 +11,35 @@ export default class ZoomAccount extends BaseModelService {
     return Models.ZoomAccount.findAll(utilizedResourceClause(...args));
   }
 
-  static async findAvailaibilityDuring(id, ...args) {
-    const data = await Models.ZoomAccount.findOne({
-      where: { id },
-      ...utilizedResourceClause(...args),
-    });
+  static async findAllAvailabilityDuring(keys) {
+    const { id, ...args } = keys[0];
+    return Models.ZoomAccount.findAll({
+      where: {
+        id: {
+          [Models.Sequelize.Op.in]: keys.map(({ id }) => id),
+        },
+      },
+      ...utilizedResourceClause(args.dateTimeRange, args.options),
+    }).then((accounts) =>
+      keys.map((key) => !accounts.find((account) => key.id === account.id)),
```

Comment:
can't compare just based on id. Need to do deep equate for the whole key.

Reply:
use https://www.npmjs.com/package/deep-equal

Reply:
```
key : {
  id: 28,
  dateTimeRange: {
    start_at: '2020-05-20T18:30:00.000Z',
    end_at: '2020-05-20T19:30:00.000Z'
  }
}
```

```
zoomAccount : {
    id: 28,
    email: 'Davis@gmail.com',
    created_at: 2020-05-21T06:45:50.203Z,
    updated_at: 2020-05-21T06:45:50.203Z

}
```

Reply:
I don't think we can do deep equate here?

Reply:
You also get ZoomAccount.resource having start_at and end_at. Form the same shape with the data from db and then equate them.